File: internal/forms/inputs/f1099nec.go
package inputs

import "taxpilot/internal/forms"

func init() { forms.RegisterForm(F1099NEC) }

// F1099NEC returns the FormDef for a 1099-NEC Nonemployee Compensation.
// 1099-NEC forms are issued by US clients and companies only.
// Foreign freelance/contractor income is entered as foreign self-employment income.
func F1099NEC() *forms.FormDef {
	return &forms.FormDef{
		ID:           forms.Form1099NEC,
		Name:         "1099-NEC Nonemployee Compensation (US payers only)",
		Jurisdiction: forms.Federal,
		TaxYears:      []int{2025},
		QuestionGroup: forms.GroupIncome1099,
		QuestionOrder: 3,
		Fields: []forms.FieldDef{
			{
				Line:   forms.LinePayerName,
				Type:   forms.UserInput,
				ValueType: forms.StringValue,
				Label:  "Payer name",
				Prompt: "What is the US payer's name (from 1099-NEC)? (Skip this form if the payer is foreign — foreign contractor income is entered separately)",
			},
			{
				Line:   forms.LinePayerTIN,
				Type:   forms.UserInput,
				ValueType: forms.StringValue,
				Label:  "Payer TIN",
				Prompt: "What is the payer's TIN (XX-XXXXXXX)?",
			},
			{
				Line:   "nonemployee_compensation",
				Type:   forms.UserInput,
				Label:  "Box 1: Nonemployee compensation",
				Prompt: "Enter Box 1 — Nonemployee compensation:",
			},
			{
				Line:   "federal_tax_withheld",
				Type:   forms.UserInput,
				Label:  "Box 4: Federal income tax withheld",
				Prompt: "Enter Box 4 — Federal income tax withheld:",
			},
		},
	}
}

File: api/src/forms/inputs/f1099_nec.rs
use crate::domain::builders::*;
use crate::domain::field::*;
use crate::domain::form::*;

/// 1099-NEC Nonemployee Compensation (US payers only).
///
/// 1099-NEC forms are issued by US clients and companies only.
/// Foreign freelance/contractor income is entered as foreign self-employment income.
/// Instance-keyed: fields are prefixed with "1099nec:1:", "1099nec:2:", etc. at runtime.
pub fn form_1099_nec() -> FormDef {
    FormDef {
        id: FORM_1099_NEC.to_string(),
        name: "1099-NEC Nonemployee Compensation (US payers only)".to_string(),
        jurisdiction: Jurisdiction::Federal,
        tax_years: vec![2024, 2025, 2026],
        question_group: "income_1099".to_string(),
        question_order: 3,
        fields: vec![
            string_input_field(
                "payer_name",
                "Payer name",
                "What is the US payer's name (from 1099-NEC)? (Skip this form if the payer is \
                 foreign -- foreign contractor income is entered separately)",
            ),
            string_input_field(
                "payer_tin",
                "Payer TIN",
                "What is the payer's TIN (XX-XXXXXXX)?",
            ),
            input_field(
                "nonemployee_compensation",
                "Box 1: Nonemployee compensation",
                "Enter Box 1 -- Nonemployee compensation:",
            ),
            input_field(
                "federal_tax_withheld",
                "Box 4: Federal income tax withheld",
                "Enter Box 4 -- Federal income tax withheld:",
            ),
        ],
    }
}
